### ISSUE TITLE
Improvements / Clarifications to process.* fields

### ISSUE THREAD

**Issue by MikePaquette** [elastic/ecs#278]
I received this feedback from some potential users:

> I've checked model for process and couldn't find "commandline", "image" and "parent commandline" fields. These are basics...
> 
> Actually, you have both image and commandline fields for process execution:
> - process.executable — that's how you named image (from sysmon)
> - process.args — that's how you named command line arguments
>  
> It would be better to have such explanation in description of fields, because it's not self-explanatory, let’s say, it’s not obvious from the first sight.

**Comment by webmat** [comment]
They want the description of `process.executable` to use the terminology "image"? This seems like a very odd term, to describe the path to an executable. Looking at the sysmon documentation, I can plainly see that the tool seems to use the word "image" to mean the path to an executable or a file. Although why Sysmon went with that terminology escapes me.

I agree that `process.args` is not a great name. It goes against our principle of using full words instead of contractions, as much as possible. And to me, "args" doesn't sound as strong an industry convention to override this principle (not as strong as it was for IP or PID).

The description can indeed be clarified a bit quite easily. That should be straightforward.

The "parent commandline" field sounds a bit contrived. If one has the parent's PID, they can go figure out the parent's command line arguments with a second lookup. If they really want all of their events enriched with this information, they're free to do the enrichment too, and add that to a custom field. But I'm not convinced we should add this to ECS just yet.

**Comment by yoda-sec** [comment]
Adding various parent process field names to ECS would have a very large benefit for security use cases. In addition to the existing ```process.ppid```, it would also be useful to somehow map in the ```event_data.ParentImage``` and ```event_data.ParentCommandLine``` fields (using Sysmon fields parsed with WinLogBeat as the example) as these are commonly used for alerting and correlation purposes. Also, different endpoint tools have various naming conventions for these parent fields, so having a standard to map would be helpful.

**Comment by andrewkroh** [comment]
I agree we need to define a place for information about the parent process. So far I’ve been nesting the process fields under `process.parent.*`. It would also be good to address the process GUID fields that are Windows specific.

For Sysmon I’ve been experimenting with mapping the fields to ECS by using a processor in Winlogbeat.
https://gist.github.com/andrewkroh/9e4c3bef0adf7b87f3ad6e54c3f4d89f#file-winlogbeat-sysmon-js-L10

**Comment by webmat** [comment]
Yes, I agree with the idea of nesting the `process` field set at `process.parent`. We should indeed be clarifying that.

@yoda-sec 2 questions for you:

* If approach this as described above, do you see any fields missing to `process` otherwise?
* Do you think "process.executable" works well for the Windows concept of "image"? I'd be curious to hear your opinion, if not.

**Comment by yoda-sec** [comment]
👍 to the ```process.parent.*``` approach as well. In addition to the GUID fields, one of the other main Sysmon fields that sticks out as missing is ```event_data.Hashes```.  

As for the '''process.executable" field, I think it works fine and is easy to understand based on the field description. However, since the description (which could be changed) calls out that it's the "absolute path" of the executable, would maybe just using ```process.path``` fit better? Going off of the above discussion about ```process.args``` not being a great name, you could maybe do something similar to the below Sysmon mapping example (again using Winlogbeat), which should be generic enough for process information from other endpoint tools and operating systems. 

```
"event_data": {
  "Image": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
  "CommandLine": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe  -NoProfile -NonInteractive -NoLogo -WindowStyle hidden -ExecutionPolicy Unrestricted \"Import-Module \"C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Module\\Ec2Launch.psd1\"; Set-Wallpaper\""
}
```
to
```
"process": {
  "path": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
  "command": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe  -NoProfile -NonInteractive -NoLogo -WindowStyle hidden -ExecutionPolicy Unrestricted \"Import-Module \"C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Module\\Ec2Launch.psd1\"; Set-Wallpaper\""
}
```
